internal/storage/image.go

```diff
@@ -365,44 +397,258 @@ func (svc *imageService) prepareReference(inputSystemContext *types.SystemContex
 }
 
 func (svc *imageService) PrepareImage(inputSystemContext *types.SystemContext, imageName string) (types.ImageCloser, error) {
-	systemContext, srcRef, err := svc.prepareReference(inputSystemContext, imageName)
+	systemContext, srcRef, err := svc.lookup.prepareReference(inputSystemContext, imageName)
 	if err != nil {
 		return nil, err
 	}
 
 	return srcRef.NewImage(svc.ctx, systemContext)
 }
 
-func (svc *imageService) PullImage(systemContext *types.SystemContext, imageName string, inputOptions *copy.Options) (types.ImageReference, error) {
-	policy, err := signature.DefaultPolicy(systemContext)
+// nolint: gochecknoinits
+func init() {
+	reexec.Register("crio-copy-image", copyImageChild)
+}
+
+type copyImageArgs struct {
+	Lookup         *imageLookupService
+	ImageName      string
+	ParentCgroup   string
+	SystemContext  *types.SystemContext
+	Options        *ImageCopyOptions
+	HasCollectMode bool
+
+	StoreOptions storage.StoreOptions
+}
+
+// moveSelfToCgroup moves the current process to a new transient cgroup.
+func moveSelfToCgroup(cgroup string, hasCollectMode bool) error {
```

Comment:
we seem to be hard coding assumptions about the node using systemd. Will this break users who aren't (if there are any?)

Reply:
yes, currently only systemd is supported.  If CRI-O is running without systemd, then we wouldn't even get this far but fail earlier.
cgroupfs support is missing, we'll have to take care of cleaning up the cgroup by ourselves.